conn2res/connectivity.py

```diff
@@ -39,21 +41,40 @@ def __init__(self, filename=None, subj_id=0, w=None, modules=None):
             else:
                 self.w = load_file('connectivity.npy')
 
-            # select one subject
-            self.w = self.w[:, :, subj_id]
+            # select one subject from group connectivity data
+            if subj_id is not None and self.w.ndim == 3:
+                self.w = self.w[:, :, subj_id]
 
         # set zero diagonal
         np.fill_diagonal(self.w, 0)
 
         # remove inf and nan
         self.w[np.logical_or(np.isinf(self.w), np.isnan(self.w))] = 0
 
+        # make sure weights are float
+        self.w = self.w.astype(float)
+
         # number of all active nodes
         self.n_nodes = len(self.w)
 
         # number of edges (in symmetric networks edges are counted twice!)
         self.n_edges = np.sum(self.w != 0)
 
+        # check if network is symmetric (needed e.g. for checking connectedness)
+        self.symmetric = check_symmetric(self.w)
+
+        # use fixed density if set
+        if density is not None:
+            if self.symmetric:
+                nedges = int(self.n_nodes * (self.n_nodes - 1) * density // 2)
+                id_ = np.argsort(np.triu(self.w, 1), axis=None)
+                self.w[np.unravel_index(id_[:-nedges], self.w.shape)] = 0
+                self.w = make_symmetric(self.w, copy_lower=False)
+            else:
+                nedges = int(self.n_nodes * (self.n_nodes - 1) * density)
+                id_ = np.argsort(self.w, axis=None)
+                self.w[np.unravel_index(id_[:-nedges], self.w.shape)] = 0
```

Comment:
I added this option to allow for adjusting the density of networks slightly (e.g. to enable comparing networks unconfounded by density) but it should be used with caution as it might make connected networks unconnected